### PR TITLE
Make sure row_event happens before save_event

### LINKED ISSUE #5: Ensure that we handle "on_row_event" before "on_save_position"
I think we do now, but we need tests to confirm it and ensure that we don't break it in the future.

### PR DESCRIPTION
**Pull request by roback**
close #5

### CODE REVIEW

File: spec/test_helper.rb [hunk 1]
```diff
@@ -24,6 +24,9 @@ def self.wait_for_row_events(ecco_client, count: 1, &block)
   end
 
   def self.wait_for_save_position_events(ecco_client, count: 1, &block)
+    # Take into account the initial save event that is emitted when client starts
+    count += 1
```
Comment (walro) [inline spec/test_helper.rb:28]: What happens if I call this method twice in the same test? Won't it "overstep"?

Comment (roback) [inline spec/test_helper.rb:28]: Nope, the client is started and stopped each time you call the method.

Comment (walro) [inline spec/test_helper.rb:28]: Ok! Because we start and stop here, got it.

File: spec/integration/client_spec.rb [hunk 2]
```diff
@@ -60,6 +60,30 @@
...
+        subject.on_row_event { |_| event_order << :on_row_event }
+        subject.on_save_position { |_, _| event_order << :on_save_position }
+
+        subject.start_in_thread
+
+        DatabaseHelper.insert(table_name, mysql_row)
+
+        sleep 0.1 while event_order.count < events_to_wait_for
+        subject.stop
+        event_order.last(2)
```
Comment (walro) [inline spec/integration/client_spec.rb:78]: Perhaps a comment why we only care about the last 2 (and why we need to wait 4 times).

**Comment by walro** [comment]
Left a comment concerning a comment, otherwise it looks good!

**Comment by walro** [comment]
LGTM, I think we can release as `0.4.1`?

**Comment by roback** [comment]
`0.4.1` sounds good since there are no new features.

**Pull request merged**